src/Foundation/NSMetadataItem.cs

```diff
@@ -18,12 +18,34 @@ bool GetBool (NSString key)
 			return n == null ? false : n.BoolValue;
 		}
 
+		bool? GetNullableBool (NSString key)
+		{
+			var n = Runtime.GetNSObject<NSNumber> (GetHandle (key));
+			return n?.BoolValue;
+		}
+
 		double GetDouble (NSString key)
 		{
 			var n = Runtime.GetNSObject<NSNumber> (GetHandle (key));
 			return n == null ? 0 : n.DoubleValue;
 		}
 
+		double? GetNullableDouble (NSString key)
+		{
+			var n = Runtime.GetNSObject<NSNumber> (GetHandle (key));
+			return n?.DoubleValue;
+		}
+
+		nint? GetNInt (NSString key)
```

Comment:
private and they should all return nullable (for value types) since the presence is optional and out of our control (breaking change until xamcore_4_0)